### ISSUE TITLE
Expo Router won't work with Safari 15

### ISSUE THREAD

**Issue by nzapponi** [expo/expo#33224]
### Minimal reproducible example

N/A

### Which package manager are you using? (Yarn is recommended)

yarn

### If the issue is web-related, please select the bundler (`web.bundler` in the `app.json`)

metro

### Summary

When compiling for web, Expo Router seems to rely on lookbehind regex, which is not supported by Safari 15. That's a relatively recent version of Safari, still used by a non-insignificant user population.

https://github.com/expo/expo/blob/5aaedd5bfa32c3e06ceb8854c92325822ba137bb/packages/expo-router/src/matchers.tsx#L25

Given the complexity in polyfilling, is there any way to either avoid using it, or directions on how to polyfill it?

Thanks

### Environment

```text
expo-env-info 1.2.1 environment info:
    System:
      OS: macOS 15.0.1
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 20.8.0 - ~/.nvm/versions/node/v20.8.0/bin/node
      npm: 10.2.0 - ~/.nvm/versions/node/v20.8.0/bin/npm
      Watchman: 2024.10.14.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.15.2 - /Users/nzapponi/.gem/ruby/2.7.7/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 24.1, iOS 18.1, macOS 15.1, tvOS 18.1, visionOS 2.1, watchOS 11.1
    IDEs:
      Android Studio: 2024.1 AI-241.18034.62.2412.12266719
      Xcode: 16.1/16B40 - /usr/bin/xcodebuild
    Expo Workflow: bare
```

**Comment by vlsido** [comment]
I get blank page with the same error on an iPhone (iOS 16.1.1) when debugging using Safari.

_"SyntaxError: Invalid regular expression: invalid group specifier name"_

The error points to the same line of code mentioned in the original post

Environment:
- "web.bundler = metro"
- Switching to "web.output = static" allows the page to render, but:
  1. The same error persists in the console
  2. Assets fail to load (not sure if this is related)

The behavior is consistent across different browsers on iPhone (tested on Safari, Chrome, Opera)

Observations:
- Not an issue on Android, PC/macOS browsers

System:
- macOS 14.5
- iOS 16.1.1

**Comment by Antonio-Serrat** [comment]
Seems that on safari is not supported [lookbehind](https://caniuse.com/js-regexp-lookbehind) until 16.4

**Comment by gitteraz** [comment]
I get blank page with the same error on an Iphone and mac when using Safari 15.
"bundler": "metro",
"output": "single"

**Comment by marklawlor** [comment]
Fixed in 4.0.11
